### PR TITLE
refactor: splitting the client/codebase into components

### PR DESCRIPTION
**Pull request by tombuildsstuff**
This PR refactors the codebase into components, since there's a number of these which are subtly different - having these in a single file is going to become problematic as this extends out.

This PR also introduces the concept of a `cleaner`, for removing items within an item, for example removing the Locks within a Resource Group (or ensuring that any ServiceBus Namespace within the Resource Group has any pairing broken, etc) - this is something touched upon in #6 which we'll want to extend, hence this foundation/plumbing

### CODE REVIEW

**Review by jackofallops — APPROVED** [review]
Just a couple questions below, but otherwise LGTM 👍

File: dalek/cleaners/subscriptions_soft_delete_managed_hsms.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+var _ SubscriptionCleaner = purgeSoftDeletedKeyVaultsInSubscriptionCleaner{}
+
+type purgeSoftDeletedKeyVaultsInSubscriptionCleaner struct {
+}
+
+func (p purgeSoftDeletedKeyVaultsInSubscriptionCleaner) Name() string {
+	return "Purging Soft Deleted Key Vaults in Subscription"
+}
+
+func (p purgeSoftDeletedKeyVaultsInSubscriptionCleaner) Cleanup(ctx context.Context, subscriptionId commonids.SubscriptionId, client *clients.AzureClient, opts options.Options) error {
```
[the remark this answers is not in the document]
Comment (tombuildsstuff) [reply dalek/cleaners/subscriptions_soft_delete_managed_hsms.go:23]: 👍 changing to `purgeSoftDeletedManagedHSMsInSubscriptionCleaner`

File: dalek/management_groups.go [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+		if props := group.Properties; props != nil && props.DisplayName != nil && d.opts.Prefix != "" {
+			if !strings.HasPrefix(strings.ToLower(*props.DisplayName), strings.ToLower(d.opts.Prefix)) {
+				continue
+			}
+		}
+
+		groupName := *group.Name
+		id := commonids.NewManagementGroupID(*group.Id)
+
+		if _, err := uuid.ParseUUID(groupName); err != nil {
```
[the remark this answers is not in the document]
Comment (tombuildsstuff) [reply dalek/management_groups.go:45]: the `groupName` here will always be the UUID, the `name` would be the display name

[the remark this answers is not in the document]
Comment (tombuildsstuff) [reply dalek/management_groups.go:45]: fixing this in a follow

**Pull request merged**